src/ckeditor.jsx

```diff
@@ -1,23 +1,23 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
 import React from 'react';
 import PropTypes from 'prop-types';
 
 export default class CKEditor extends React.Component {
-
 	constructor( props ) {
 		super( props );
 
-		this.editorInstance = null;
-	}
-
-	// This component should never be updated by React itself.
-	shouldComponentUpdate() {
-		return false;
+		// After mounting the editor, the variable will contain a reference to created editor.
+		// @see: https://docs.ckeditor.com/ckeditor5/latest/api/module_core_editor_editor-Editor.html
+		this.editor = null;
 	}
 
-	// Update editor data if data property is changed.
-	componentWillReceiveProps( newProps ) {
-		if ( this.editorInstance && newProps.data ) {
-			this.editorInstance.setData( newProps.data );
+	componentDidUpdate( newProps ) {
+		if ( this.editor && newProps.data ) {
```

Comment:
I'm thinking about cases when someone stores editor content in the external place.

Then it'd look like this:

editor changes => ckeditor component changes => parent component saves the data => some redux store detects changes => redux store updates => component updates with same props => editor changes. And it'd create an infinite loop for now.

I'd write it like this:
```js
componentDidUpdate() {
	if ( this.editor && this.editor.getData() !== this.props.data ) {
		this.editor.setData( this.props.data );
	}
}
```

It should be safer.

Reply:
React + redux is pretty common. And people would need to hack to prevent component updates.

> Except that, this.editor.getData() never returns an empty string. If this.props.data will be empty, this if will always return true.

Right, but once the props change to `<p>&nbsp;</p>` the cycle ends, because the `props.data` and `editor.getData()` will return the same string.

Reply:
> React + redux is pretty common. And people would need to hack to prevent component updates.

Would be awesome to see almost one ticket which explains the problem. In fact, I'm making my first steps with React and I would like more opinions. At this moment, people want to use the package and they cannot. That's the goal of the PR. Publish and allows using (and testing). If something will be broken, we will fix that. Now, I cannot imagine the whole picture.

> Right, but once the props change to <p>&nbsp;</p> the cycle ends, because the props.data and editor.getData() will return the same string.

Sure? Are the properties bound two-ways? Component => CKEditor will update the content, but content updated by a user will not change the Component property. Maybe I'm wrong. I don't insist.